tests/src/test/java/org/eclipse/hono/tests/client/KafkaProducerHelperIT.java

```diff
@@ -0,0 +1,190 @@
+ * e.g. <code>-Ddownstream.bootstrap.servers="PLAINTEXT://localhost:9092"</code>.
+ */
+@ExtendWith(VertxExtension.class)
+@AssumeMessagingSystem(type = MessagingType.kafka)
+public class KafkaProducerHelperIT {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaProducerHelperIT.class);
+
+    private static final short REPLICATION_FACTOR = 1;
+
+    private static Vertx vertx;
+    private static KafkaAdminClient adminClient;
+    private static List<String> topicsToDeleteAfterTests;
+
+    private KafkaProducer<String, Buffer> kafkaProducer;
+
+    /**
+     * Sets up fixture.
+     */
+    @BeforeAll
+    public static void init() {
+        vertx = Vertx.vertx();
+        topicsToDeleteAfterTests = new ArrayList<>();
+
+        final Map<String, String> adminClientConfig = IntegrationTestSupport.getKafkaAdminClientConfig()
+                .getAdminClientConfig("test");
+        adminClient = KafkaAdminClient.create(vertx, adminClientConfig);
+    }
+
+    /**
+     * Creates the Kafka producer.
+     */
+    @BeforeEach
+    void initProducer() {
+        final Map<String, String> producerConfig = IntegrationTestSupport.getKafkaProducerConfig()
+                .getProducerConfig("test");
+        kafkaProducer = KafkaProducer.create(vertx, producerConfig);
+    }
+
+    /**
+     * Cleans up fixture.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @AfterAll
+    public static void shutDown(final VertxTestContext ctx) {
+        final Promise<Void> topicsDeletedPromise = Promise.promise();
+        adminClient.deleteTopics(topicsToDeleteAfterTests, topicsDeletedPromise);
+        topicsDeletedPromise.future()
+                .recover(thr -> {
+                    LOG.info("error deleting topics", thr);
+                    return Future.succeededFuture();
+                })
+                .onComplete(ar -> {
+                    topicsToDeleteAfterTests.clear();
+                    topicsToDeleteAfterTests = null;
+                    adminClient.close();
+                    adminClient = null;
+                    vertx.close();
+                    vertx = null;
+                })
+                .onComplete(ctx.succeedingThenComplete());
+    }
+
+    /**
+     * Closes the Kafka producer created during the test.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @AfterEach
+    void closeProducer(final VertxTestContext ctx) {
+        if (kafkaProducer != null) {
+            kafkaProducer.close().onComplete(ctx.succeedingThenComplete());
+        }
+    }
+
+    /**
+     * Verifies that removing topic-specific metrics in a Kafka producer is working as expected.
+     *
+     * @throws InterruptedException if test execution gets interrupted.
+     */
+    @Test
+    @Timeout(value = 10, timeUnit = TimeUnit.SECONDS)
+    public void testProducerTopicMetricsGetRemoved() throws InterruptedException {
```

Comment:
wouldn't it make more sense to test this in the context of actually deleting a tenant?

Reply:
It's a test with a different scope then. But I'm fine with adding a combined integration test like that. I've created `KafkaBasedEventSenderIT` now (and removed `KafkaProducerHelperIT`).